### PR TITLE
Make e-commerce demo canonical 

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Invoices
- [x] Orders
- [x] Finishing touches
- [x] Fix UserMenu on small screens
- [x] Fix AutocompleteInput doesn't have a minimum width (and doesn't support `sx`)

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx [hunk 1]
```diff
@@ -517,6 +519,21 @@ If you provided a React element for the optionText prop, you must also provide t
...
+};
+
+const StyledAutocomplete = styled(Autocomplete, {
+    name: PREFIX,
+    overridesResolver: (props, styles) => styles.root,
+})(({ theme }) => ({
+    [`& .${AutocompleteInputClasses.textField}`]: {
+        minWidth: theme.spacing(20),
+    },
+}));
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:535]: You have to update the upgrade guide and the docs about this.

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by WiXSL — CHANGES_REQUESTED** [review]
Customers: In the following gif you can see two issues, The moving inputs when selecting items and the cut of avatar on the app bar

chrome-capture (1)

**Review by djhi — APPROVED** [review]
:fire:

File: packages/ra-core/src/dataProvider/useIsDataLoaded.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * Check if react-query has already fetched data for a query key.
+ *
+ * This hook is reactive.
+ *
+ * @example
+ * const isCustomerLoaded = useIsDataLoaded(['customers', 'getOne', { id: customerId }]);
+ *
+ * @returns {boolean} true if the data is loaded, false otherwise
+ */
+export const useIsDataLoaded = (
```
Comment (djhi) [inline packages/ra-core/src/dataProvider/useIsDataLoaded.ts:14]: It would be nice to document this

**Pull request merged**